File: internal/installer/providers.go
package installer

// ProviderInfo is what the UI sees in /api/providers.
type ProviderInfo struct {
	ID             ProviderID `json:"id"`
	Label          string     `json:"label"`
	Description    string     `json:"description"`
	BaseURL        string     `json:"baseUrl,omitempty"`
	Type           string     `json:"type"`
	DefaultModel   string     `json:"defaultModel"`
	ModelsEndpoint string     `json:"modelsEndpoint,omitempty"`
	IsDefault      bool       `json:"isDefault,omitempty"`
	DocsURL        string     `json:"docsUrl"`
}

// Providers lists the LLM backends supported by the wizard. tensorix is
// pinned first with IsDefault=true so the UI surfaces it as the primary
// option without doing client-side ordering tricks.
var Providers = []ProviderInfo{
	{
		ID:             ProviderTensorix,
		Label:          "Tensorix",
		Description:    "Default. Unified OpenAI-compatible gateway with curated open-weights and proprietary models.",
		BaseURL:        "https://api.tensorix.ai/v1",
		ModelsEndpoint: "https://api.tensorix.ai/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "tensorix/glm-4.6",
		IsDefault:      true,
		DocsURL:        "https://docs.tensorix.ai",
	},
	{
		ID:           ProviderAnthropic,
		Label:        "Anthropic",
		Description:  "Claude Opus, Sonnet, Haiku — top-tier reasoning and tool use.",
		Type:         "native",
		DefaultModel: "anthropic/claude-opus-4-7",
		DocsURL:      "https://docs.anthropic.com",
	},
	{
		ID:             ProviderOpenAI,
		Label:          "OpenAI",
		Description:    "GPT family + o-series reasoning models.",
		BaseURL:        "https://api.openai.com/v1",
		ModelsEndpoint: "https://api.openai.com/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "openai/gpt-4o",
		DocsURL:        "https://platform.openai.com/docs",
	},
	{
		ID:           ProviderGoogle,
		Label:        "Google Gemini",
		Description:  "Gemini Pro / Flash — strong multimodal and long context.",
		Type:         "native",
		DefaultModel: "google/gemini-2.5-pro",
		DocsURL:      "https://ai.google.dev",
	},
	{
		ID:             ProviderMistral,
		Label:          "Mistral",
		Description:    "Mistral Large, Codestral and embedding models.",
		BaseURL:        "https://api.mistral.ai/v1",
		ModelsEndpoint: "https://api.mistral.ai/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "mistral/mistral-large-latest",
		DocsURL:        "https://docs.mistral.ai",
	},
	{
		ID:             ProviderDeepSeek,
		Label:          "DeepSeek",
		Description:    "DeepSeek-V3, DeepSeek-R1 reasoning model.",
		BaseURL:        "https://api.deepseek.com/v1",
		ModelsEndpoint: "https://api.deepseek.com/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "deepseek/deepseek-chat",
		DocsURL:        "https://api-docs.deepseek.com",
	},
	{
		ID:             ProviderGroq,
		Label:          "Groq",
		Description:    "Ultra-low-latency inference for Llama, Mixtral, Qwen.",
		BaseURL:        "https://api.groq.com/openai/v1",
		ModelsEndpoint: "https://api.groq.com/openai/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "groq/llama-3.3-70b-versatile",
		DocsURL:        "https://console.groq.com/docs",
	},
	{
		ID:             ProviderTogether,
		Label:          "Together AI",
		Description:    "Hosted open-weights models with OpenAI compatibility.",
		BaseURL:        "https://api.together.xyz/v1",
		ModelsEndpoint: "https://api.together.xyz/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "together/meta-llama/Llama-3.3-70B-Instruct-Turbo",
		DocsURL:        "https://docs.together.ai",
	},
	{
		ID:           ProviderPerplexity,
		Label:        "Perplexity",
		Description:  "Search-grounded LLMs (sonar family).",
		BaseURL:      "https://api.perplexity.ai",
		Type:         "openai-compatible",
		DefaultModel: "perplexity/sonar-pro",
		DocsURL:      "https://docs.perplexity.ai",
	},
	{
		ID:             ProviderXAI,
		Label:          "xAI",
		Description:    "Grok family.",
		BaseURL:        "https://api.x.ai/v1",
		ModelsEndpoint: "https://api.x.ai/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "xai/grok-2",
		DocsURL:        "https://docs.x.ai",
	},
	{
		ID:             ProviderOllama,
		Label:          "Ollama (local)",
		Description:    "Run open models locally; no API key required.",
		BaseURL:        "http://localhost:11434/v1",
		ModelsEndpoint: "http://localhost:11434/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "ollama/llama3.2",
		DocsURL:        "https://ollama.com",
	},
	{
		ID:             ProviderLMStudio,
		Label:          "LM Studio (local)",
		Description:    "Local desktop runtime with OpenAI-compatible server.",
		BaseURL:        "http://localhost:1234/v1",
		ModelsEndpoint: "http://localhost:1234/v1/models",
		Type:           "openai-compatible",
		DefaultModel:   "lmstudio/llama-3.2-3b-instruct",
		DocsURL:        "https://lmstudio.ai",
	},
}

func FindProvider(id ProviderID) (ProviderInfo, bool) {
	for _, p := range Providers {
		if p.ID == id {
			return p, true
		}
	}
	return ProviderInfo{}, false
}

File: packages/installer-server/src/lib/providers.ts
import type { ProviderId } from '../schema/openclaw.schema.js';

export interface ProviderInfo {
  id: ProviderId;
  label: string;
  description: string;
  baseUrl?: string;
  type: 'openai-compatible' | 'native';
  defaultModel: string;
  modelsEndpoint?: string;
  isDefault?: boolean;
  docsUrl: string;
}

export const PROVIDERS: ProviderInfo[] = [
  {
    id: 'tensorix',
    label: 'Tensorix',
    description: 'Default. Unified OpenAI-compatible gateway with curated open-weights and proprietary models.',
    baseUrl: 'https://api.tensorix.ai/v1',
    modelsEndpoint: 'https://api.tensorix.ai/v1/models',
    type: 'openai-compatible',
    defaultModel: 'tensorix/glm-4.6',
    isDefault: true,
    docsUrl: 'https://docs.tensorix.ai',
  },
  {
    id: 'anthropic',
    label: 'Anthropic',
    description: 'Claude Opus, Sonnet, Haiku — top-tier reasoning and tool use.',
    type: 'native',
    defaultModel: 'anthropic/claude-opus-4-7',
    docsUrl: 'https://docs.anthropic.com',
  },
  {
    id: 'openai',
    label: 'OpenAI',
    description: 'GPT family + o-series reasoning models.',
    baseUrl: 'https://api.openai.com/v1',
    modelsEndpoint: 'https://api.openai.com/v1/models',
    type: 'openai-compatible',
    defaultModel: 'openai/gpt-4o',
    docsUrl: 'https://platform.openai.com/docs',
  },
  {
    id: 'google',
    label: 'Google Gemini',
    description: 'Gemini Pro / Flash — strong multimodal and long context.',
    type: 'native',
    defaultModel: 'google/gemini-2.5-pro',
    docsUrl: 'https://ai.google.dev',
  },
  {
    id: 'mistral',
    label: 'Mistral',
    description: 'Mistral Large, Codestral and embedding models.',
    baseUrl: 'https://api.mistral.ai/v1',
    modelsEndpoint: 'https://api.mistral.ai/v1/models',
    type: 'openai-compatible',
    defaultModel: 'mistral/mistral-large-latest',
    docsUrl: 'https://docs.mistral.ai',
  },
  {
    id: 'deepseek',
    label: 'DeepSeek',
    description: 'DeepSeek-V3, DeepSeek-R1 reasoning model.',
    baseUrl: 'https://api.deepseek.com/v1',
    modelsEndpoint: 'https://api.deepseek.com/v1/models',
    type: 'openai-compatible',
    defaultModel: 'deepseek/deepseek-chat',
    docsUrl: 'https://api-docs.deepseek.com',
  },
  {
    id: 'groq',
    label: 'Groq',
    description: 'Ultra-low-latency inference for Llama, Mixtral, Qwen.',
    baseUrl: 'https://api.groq.com/openai/v1',
    modelsEndpoint: 'https://api.groq.com/openai/v1/models',
    type: 'openai-compatible',
    defaultModel: 'groq/llama-3.3-70b-versatile',
    docsUrl: 'https://console.groq.com/docs',
  },
  {
    id: 'together',
    label: 'Together AI',
    description: 'Hosted open-weights models with OpenAI compatibility.',
    baseUrl: 'https://api.together.xyz/v1',
    modelsEndpoint: 'https://api.together.xyz/v1/models',
    type: 'openai-compatible',
    defaultModel: 'together/meta-llama/Llama-3.3-70B-Instruct-Turbo',
    docsUrl: 'https://docs.together.ai',
  },
  {
    id: 'perplexity',
    label: 'Perplexity',
    description: 'Search-grounded LLMs (sonar family).',
    baseUrl: 'https://api.perplexity.ai',
    type: 'openai-compatible',
    defaultModel: 'perplexity/sonar-pro',
    docsUrl: 'https://docs.perplexity.ai',
  },
  {
    id: 'xai',
    label: 'xAI',
    description: 'Grok family.',
    baseUrl: 'https://api.x.ai/v1',
    modelsEndpoint: 'https://api.x.ai/v1/models',
    type: 'openai-compatible',
    defaultModel: 'xai/grok-2',
    docsUrl: 'https://docs.x.ai',
  },
  {
    id: 'ollama',
    label: 'Ollama (local)',
    description: 'Run open models locally; no API key required.',
    baseUrl: 'http://localhost:11434/v1',
    modelsEndpoint: 'http://localhost:11434/v1/models',
    type: 'openai-compatible',
    defaultModel: 'ollama/llama3.2',
    docsUrl: 'https://ollama.com',
  },
  {
    id: 'lmstudio',
    label: 'LM Studio (local)',
    description: 'Local desktop runtime with OpenAI-compatible server.',
    baseUrl: 'http://localhost:1234/v1',
    modelsEndpoint: 'http://localhost:1234/v1/models',
    type: 'openai-compatible',
    defaultModel: 'lmstudio/llama-3.2-3b-instruct',
    docsUrl: 'https://lmstudio.ai',
  },
];

export function findProvider(id: ProviderId): ProviderInfo {
  const p = PROVIDERS.find((x) => x.id === id);
  if (!p) throw new Error(`Unknown provider: ${id}`);
  return p;
}
